File: 540-single-element-in-a-sorted-array/single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        HashMap<Integer, Integer>freq = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            freq.put(nums[i], freq.getOrDefault(nums[i], 0)+1);
        }
        for(Map.Entry<Integer, Integer> it : freq.entrySet()){
            if(it.getValue() == 1)
            return it.getKey();
        }
        return -1; 
    }
}

File: 540-single-element-in-a-sorted-array/single-element-in-a-sorted-array.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        if(nums.size() == 1) return nums[0];
        if(nums[0] != nums[1]) return nums[0];
        if(nums[nums.size()-1] != nums[nums.size()-2]) return nums[nums.size()-1];
        int l = 1;
        int h = nums.size()-2;
        while(l <= h){
            int mid = (l+h)/2;
            if(nums[mid] != nums[mid-1] && nums[mid] != nums[mid+1])
            return nums[mid];
            if(mid%2 != 0){
                if(nums[mid] == nums[mid-1]){
                    l = mid + 1;
                }else{
                    h = mid - 1;
                }
            }else{
                if(nums[mid] == nums[mid-1]){
                    h = mid - 1;
                }else{
                    l = mid + 1;
                }
            }
        }
        return -1;
    }
};